Reading List/Reading List/Controllers/BookController.swift

```diff
@@ -0,0 +1,145 @@
+Add a "Create" method that initializes a new Book object. In order to persist the newly created Book, call the saveToPersistentStore() method at the end of this function.
+Add a "Delete" method that passes in a Book object as a parameter, and removes it from the books array.
+We'll need two "Update" methods:
+One is to update a Book object's hasBeenRead property. Call it updateHasBeenRead(for book: Book). It should simply swap the hasBeenRead value from false to true and vice-versa.
+The other is to edit the Book's title and/or reasonToRead properties.
+
+You may have noticed from the screen recording that the table view is going to have more than one section. In order to facilitate the implementation of this multi-section table view, we're going to add a couple computed properties to the BookController:
+
+Create a computed property called readBooks: [Book]. Inside of the closure of the computed property, you will need to return an array of all of the Book objects from the books array whose hasBeenRead property is true. The easiest way to do that is by using the .filter higher-order function. If you are unfamiliar with this function, read the part of this article titled "Filter". If you still have questions, please ask in the help channel for your cohort, and a PM will explain it to you in more depth.
+Create a similar computed property called unreadBooks: [Book] that does the same thing, except it returns an array of Books whose hasBeenRead property is false.
+*/
+
+import Foundation
+
+class BookController {
+    
+    var books: [Book] = []
+    
+    var readingListURL: URL? {
+        let fileManager = FileManager.default
+        guard let documentsDirectory = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first else {
+            return nil
+        }
+        // Users/Angel/Documents
+        let finalLocation = documentsDirectory.appendingPathComponent("ReadingList.plist")
+        return finalLocation
+    }
+    
+    
+    func saveToPersistentStore(){
+        guard let url = readingListURL else { return } // ensuring there is a valid save destination
+        
+        let encoder = PropertyListEncoder()
+        do{
+            let data = try encoder.encode(books)
+            try data.write(to: url)
+        } catch{
+            print(error)
+        }
+    }
+    
+    func loadFromPersistentStore(){
+        
+        do{
+            guard let url = readingListURL else { return }
+            
+            let data = try Data(contentsOf: url)
+            let decoder = PropertyListDecoder()
+            let decodedBooks = try decoder.decode([Book].self, from: data)
+            books = decodedBooks
+        } catch {
+            print ("no url or no data at url")}
+    }
+    
+    func Create(title: String, reason : String){
+        let newBook = Book(title: title, reasonToRead: reason, hasBeenRead: false)
```

Comment:
If you change the initializer of Book. You don;t have to provide a `hasBeenRead` value
```suggestion
        let newBook = Book(title: title, reasonToRead: reason)
```